### PR TITLE
fix: alert modal bug fix

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY

Due to some refactoring, we were no longer updating the dropdown in the alert/report modal when a user switched from one alert option to another. 

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/Select/AsyncSelect.tsx [hunk 1]
```diff
@@ -442,7 +442,7 @@ const AsyncSelect = forwardRef(
       fetchedQueries.current.clear();
       setAllValuesLoaded(false);
       setSelectOptions(initialOptions);
-    }, [initialOptions]);
+    }, [initialOptions, options]);
```
[the remark this answers is not in the document]
Comment (AAfghahi) [reply superset-frontend/src/components/Select/AsyncSelect.tsx:445]: Yeah! So, in AlertReportModal, what we are sending over to options is not an array, it returns this:
https://github.com/apache/superset/blob/master/superset-frontend/src/views/CRUD/alert/AlertReportModal.tsx#L811

so this fails the `Array.isArray(options)` and does not update initialOptions, and leaves initialOptions as `EMPTY_OPTIONS`. This is what was causing the behavior from before.

[the remark this answers is not in the document]
Comment (michael-s-molina) [reply superset-frontend/src/components/Select/AsyncSelect.tsx:445]: I understood. I think we had a problem during the refactoring. For the async Select, we don't have initial options so we can remove the references to it. I'm sending you a patch with the changes. Let me know if `AlertReportModal` works after applying it with `git apply patch.txt`

[patch.txt](https://github.com/apache/superset/files/10396789/patch.txt)

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Pull request merged**